components/Header.js

```diff
@@ -12,13 +12,19 @@ class Header extends Component {
     this.state = { menuHidden: true };
   }
 
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    if (this.state.menuHidden !== prevState.menuHidden) {
+      this.props.onMenuToggle(this.state.menuHidden);
```

Comment:
Why does this logic need to come out of the `handleMenuToggle` function?

Reply:
Sorry, should have written a better description 🙃

The primary issue was that the old call to onMenuToggle() was sometimes happening before the state had actually been updated, since the update is not always synchronous.

According to the [setState docs](https://reactjs.org/docs/react-component.html#setstate):
> setState() does not always immediately update the component. It may batch or defer the update until later. This makes reading this.state right after calling setState() a potential pitfall. Instead, use componentDidUpdate or a setState callback (setState(updater, callback)), either of which are guaranteed to fire after the update has been applied. If you need to set the state based on the previous state, read about the updater argument below.